Blog: clean up styling, add banner for guest posts

MarkLodato opened the pull request:
Context: https://github.com/slsa-framework/slsa/issues/309#issuecomment-1113559601

Add a "views expressed disclaimer" on guest posts. 

Clean up the styling of blog posts:
- Smaller header with white background.
- Format the date to look a bit nicer.

Preview: https://deploy-preview-379--slsa.netlify.app/blog/2022/04/slsa-is-no-free-lunch

![top](https://user-images.githubusercontent.com/58860/166480624-7e97ae65-3582-44e2-a14a-23d8874fdeab.png)

[...]

![bottom](https://user-images.githubusercontent.com/58860/166481177-b791e29d-3d60-4581-9d84-f42207296614.png)


# What it looked like before

![old](https://user-images.githubusercontent.com/58860/166041870-9bfc93ba-09a2-451d-8b3b-26eb26c70083.png)

inferno-chromium reviewed (approved):
Perfect, thank you so much. I would prefer the banner to come at end of post instead of front of every post, but open to other opinions.

MarkLodato commented:
> Perfect, thank you so much. I would prefer the banner to come at end of post instead of front of every post, but open to other opinions.

My feeling is that this is important to state up front. I don't think most readers will see a footer. Also official posts don't have this banner, e.g. for announcements.

Perhaps there's a way to make it less obtrusive?

inferno-chromium commented:
> > Perfect, thank you so much. I would prefer the banner to come at end of post instead of front of every post, but open to other opinions.
> 
> My feeling is that this is important to state up front. I don't think most readers will see a footer. Also official posts don't have this banner, e.g. for announcements.
> 
> Perhaps there's a way to make it less obtrusive?

Yes these kind of banners are not common, e.g. check https://blog.sigstore.dev/, https://kubernetes.io/blog/, https://openssf.org/blog/, that is why was suggesting at end or even skip. Lets get @slsa-framework/slsa-steering-committee and community opinion on this.

lumjjb commented:
+1 on notice at the bottom.

Other ideas:
- have an additional info button beside the "authored by" which will show text when hovered
- have a thinner SLSA blogpost banner on top on the blog title with a link to more about the SLSA blog or having the text there.

MarkLodato commented:
Sounds good, moved it to the bottom which is the simplest approach. I also updated the PR to include the new SBOM article and to simplify the implementation slightly.

lumjjb reviewed (approved):
LGTM

inferno-chromium reviewed (approved):
Thank you!

kimsterv commented:
lgtm!
